peerfinder/peerfinder.py

```diff
@@ -185,60 +188,131 @@ def main():
-                row.append(line)
-            ix_tab.add_row(row)
+    for fac in common_fac_list:
+        for peer in peers:
+            curr_fac = asdict(fetch_fac_from_facilities(fac, peer.present_in))
+            if curr_fac['name'] not in common_facs:
+                common_facs.update({curr_fac['name']: {}})
+            common_facs[curr_fac['name']].update({
+                asdict(peer)['name']: asdict(peer)['ASN']
+            })
+    common_points.update({
+            "IXs": common_ixs,
+            "FACs": common_facs
+    })
+    return(common_points)
+
+def human_readable_print(common_points: Dict, peers: List):
+    """ Return always 0
+    Arguments:
+        common_points: A dict containing all common facilites and IXs
+    Returns:
+        Always zero
+    """
+
+    header = []
+    header.append("IX")
+
+    for peer in peers:
+        header.append(peer.name)
+    
+    ix_tab = PrettyTable(header)
+    ix_tab.print_empty = False
+
+    for ix in common_points['IXs']:
+        row = [ix]
+        for peer in peers:
+            
+            v4 = "v4: " + "\n".join([str(i) for i in common_points['IXs'][ix][asdict(peer)['ASN']]['IPv4']])
+            v6 = "v6: " + "\n".join(
+                [str(i) for i in common_points['IXs'][ix][asdict(peer)['ASN']]['IPv6'] if str(i) != "0100::"]
+            )
+            v6 = v6 if v6 != "v6: " else ""
+            line = f"{v4}\n{v6}"
+            row.append(line)
+        ix_tab.add_row(row)
 
         ix_tab.hrules = 1
-        print(ix_tab.get_string(sortby="IX"))
+    print(ix_tab.get_string(sortby="IX"))
 
-    if args.fac_only:
-        common_fac_list = fetch_common_facilities(peers)
-        if len(common_fac_list) < 1:
-            print("Didnt find any common Facility, exiting...")
-            exit(1)
-        header = []
-        header.append("Facility")
+    header = []
+    header.append("Facility")
+    for peer in peers:
+        header.append(peer.name)
+
+    ix_tab = PrettyTable(header)
+    ix_tab.print_empty = False
+
+    for fac in common_points['FACs']:
+        row = [fac]
+        
         for peer in peers:
-            header.append(peer.name)
+            curr_fac = common_points['FACs'][fac][asdict(peer)['name']]
+            line = f"ASN: {curr_fac}"
+            row.append(line)
+        ix_tab.add_row(row)
 
-        ix_tab = PrettyTable(header)
-        ix_tab.print_empty = False
+    ix_tab.hrules = 1
+    
+    print(ix_tab.get_string(sortby="Facility"))
 
-        for fac in common_fac_list:
-            row = [fac]
-            for peer in peers:
-                curr_fac = fetch_fac_from_facilities(fac, peer.present_in)
-                line = f"ASN: {curr_fac.ASN}"
-                row.append(line)
-            ix_tab.add_row(row)
+    return 0
 
-        ix_tab.hrules = 1
-        print(ix_tab.get_string(sortby="Facility"))
 
+def main():
+
+    args = getArgs()
+    peers = get_peers(args)
+
+    if args.ix_only and args.fac_only:
```

Comment:
and args.fac_only is not needed here, as we will fallthrough to the next elif and print the required output.